packages/twenty-server/src/modules/attachment/standard-objects/attachment.object-metadata.ts

```diff
@@ -55,6 +57,36 @@ export class AttachmentObjectMetadata extends BaseObjectMetadata {
   })
   type: string;
 
+  @FieldMetadata({
```

Comment:
Sorry for the change, but after discussion with the team, we now want to create a new composite FieldMetadataType `File` (you can take example on `FieldMetadataType.Link` in `packages/twenty-server/src/engine/metadata-modules/field-metadata/composite-types/link.composite-type.ts`) instead of creating a new storage driver type, since `gmail` is not really a storage driver type.
This composite field should have the following properties:
- **fileName**: string
- **fileExtension**: string
- **uploadedAt**: DateTime
- **storageType**: 'server' | 'gmail'
- We also want to add a virtual property `fullPath` which will be computed inside `query-result-getters.factory`.

The table `Attachment` will only have one column of type `File` (in addition to the id, createdAt, updatedAt ... and the relations).

In `packages/twenty-server/src/engine/core-modules/file/controllers/file.controller.ts` we want to create a new endpoint `@Get(gmail/:filename)` which will call [users.messages.attachments.get](https://developers.google.com/gmail/api/reference/rest/v1/users.messages.attachments/get) to get the fileStream.

@charlesBochet can help you if you have any questions regarding this change.

Reply:
Makes sense 👍 Ive actually already added one composite field :D https://github.com/twentyhq/twenty/pull/4492

Reply:
Great! It will actually be very similar, with an additional tweak on the getters logic! I can assist you if you have any question :)

Reply:
Feel free to ping on Discord direclty if you need real time inputs!